frontend/src/stats/card_recall_stats.tsx

```diff
@@ -0,0 +1,70 @@
+import { dateString } from "../date_string";
+import { CardRecallStat, ParsedFsrsCardStats } from "./types";
+import { Text, TextProps } from "@mantine/core";
+
+export interface CardRecallStatsProps {
+  card: ParsedFsrsCardStats;
+  showTime?: boolean;
+  textProps?: Exclude<TextProps, "component">;
+  valueProps?: Exclude<TextProps, "component">;
+  excludeStats?: Set<CardRecallStat>;
+}
+
+export function CardRecallStats({
+  card,
+  textProps,
+  valueProps,
+  excludeStats = new Set(),
+  showTime = false,
+}: CardRecallStatsProps) {
+  // The first time a card is incorrectly answered is not logged as a
+  // lapse, so we exclude that from the calculation of recall rate.
```

Comment:
This is not entirely true. If you answer it incorrectly after getting it right the very first time you see it, that counts as a lapse. If you answer it incorrectly the first time, and then answer it correctly the second time, you have no lapses. So subtracting 1 always is not quite right.

Reply:
Ahh I see, in that case I don't think we can get to recall % without a back-end change as well, right? probably not worth the effort then relative to other tasks I was gonna take a look at

Reply:
yeah, I don't think we can get recall % without keeping track of the actual number of times the question has been answered wrong. The best way I can think of getting this is from the log array that gets stored with each question, and just counting the incorrect entries. Or we can cache that number in that JSON blob too. But probably not worth it for right now.

Reply:
Looping back, how certain are you that this is correct?

> If you answer it incorrectly the first time, and then answer it correctly the second time, you have no lapses

I'm pretty certain I have a non-zero number of cards I've never gotten right that show `Times forgotten: {times seen - 1}`